scripts/build_awssdk.bat

```diff
@@ -4,7 +4,7 @@
 ::
 @echo off
 set aws_src_version=1.11.283
-set aws_build_version=6
+set aws_build_version=7
```

Comment:
windows 32 build fails for vs14 with:
```
2024-07-19 01:13:14  deps/aws-sdk-cpp-1.11.283/src/aws-cpp-sdk-core/source/client/RequestCompression.cpp(89): error C2220: warning treated as error - no 'object' file generated (compiling source file deps\aws-sdk-cpp-1.11.283\cmake-build-win32-VS14-Release\src\aws-cpp-sdk-core\ub_core.cpp) [deps\aws-sdk-cpp-1.11.283\cmake-build-win32-VS14-Release\src\aws-cpp-sdk-core\aws-cpp-sdk-core.vcxproj]
```
and for vs 17
```
2024-07-19 01:12:16  deps\aws-sdk-cpp-1.11.283\src\aws-cpp-sdk-core\source\client\RequestCompression.cpp(89,27): error C2220: the following warning is treated as an error [deps\aws-sdk-cpp-1.11.283\cmake-build-win32-VS17-Release\src\aws-cpp-sdk-core\aws-cpp-sdk-core.vcxproj]
2024-07-19 01:12:16    (compiling source file 'ub_core.cpp')
```

Reply:
It failed only on Win32. 
Win64, Linux and Mac are ok.

Reply:
@sfc-gh-dprzybysz @sfc-gh-dstempniak 
Tried to add some fix in `RequestCompression.cpp` but seems not working. I can't reproduce the issue with either vs14 or vs17, and vs16 build on github also works.
Please continue to share the error information. In the worst case we could revert the build flag changes for awssdk made in #731 to unblock the release. That's for binskim check which doesn't have functionality impact.

Reply:
Thanks, issue fixed.